### PR TITLE
Improve hiDPI appearance for some DM elements

### LINKED ISSUE #14817: [HiDPI] Dark mode scrollbar positions in edit/listbox controls in hiDPI wrong
Dark mode scrollbar positions in edit/listbox controls in hiDPI are in very rare case wrong.

### Steps to Reproduce the Issue

1. Have multi-monitor setup with different dpi scaling
2. Have dark mode enabled 
3. Open window with edit/listbox control which has scrollbar

note: It's is very hard to reproduce. 

### Expected Behavior

image

### Actual Behavior

dpi

### Debug Information

Notepad++ v8.6.4   (64-bit)
Build time : Feb 20 2024 - 00:12:59
Path : D:\Test\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : ON
Cloud Config : OFF
OS Name : Windows 11 Pro (64-bit)
OS Version : 23H2
OS Build : 22631.3155
Current ANSI codepage : 1252
Plugins : 
    mimeTools (3.1)
    NppConverter (4.6)
    NppExport (0.4)

### PR DESCRIPTION
**Pull request by ozone10**
fix #14817

### CODE REVIEW

**Comment by chcg** [comment]
@ozone10 Does this change the supported win versions? See https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/SUPPORTED_SYSTEM.md .

**Comment by ozone10** [comment]
@chcg 
no.
`_WIN32_WINNT=_WIN32_WINNT_WIN10 NTDDI_VERSION=NTDDI_WIN10_RS5` are needed for Windows 10 DPI APIs.
Before calling them we check that OS is supported with if clause `NppDarkMode::isWindows10()`, so there should be no issue with it.

It's possible to not use them, but it would require using undocumented APIs.

This PR is also preparation to properly support HiDPI for Windows 10/11 and newer.

**Comment by xomx** [comment]
@ozone10 
> This PR is also preparation to properly support HiDPI for Windows 10/11 and newer.

When I saw the details of this PR, I was hoping you would say something like this.
So can we look forward to PerMonitor DPI-Aware stuff etc?

**Comment by ozone10** [comment]
@xomx 
I will try to prepare and implement basic stuffs. But some elements will require more, e.g. Window dlg, docking panels tab and for some of them their contents, support for plugins, ...

**Comment by donho** [comment]
> This PR is also preparation to properly support HiDPI for Windows 10/11 and newer.

>>When I saw the details of this PR, I was hoping you would say something like this.
>>So can we look forward to PerMonitor DPI-Aware stuff etc?

>>>I will try to prepare and implement basic stuffs. But some elements will require more, e.g. Window dlg, docking panels tab and for some of them their contents, support for plugins, ...

@ozone10 
If I understand well, DPIManagerV2, which will manage better in Win11 and next version of Windows, can replace current DPI Manager in the future?

**Review by donho — CHANGES_REQUESTED** [review]
Please follow the suggestions:

File: PowerEditor/src/dpiAware.manifest [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
-         xmlns="http://schemas.microsoft.com/SMI/2005/WindowsSettings">
-      <dpiAware>true</dpiAware>
-    </asmv3:windowsSettings>
-  </asmv3:application>
+    <asmv3:application xmlns:asmv3="urn:schemas-microsoft-com:asm.v3">
+        <asmv3:windowsSettings xmlns="http://schemas.microsoft.com/SMI/2005/WindowsSettings">
+            <dpiAware>true</dpiAware>
+        </asmv3:windowsSettings>
+        <asmv3:windowsSettings xmlns="http://schemas.microsoft.com/SMI/2016/WindowsSettings">
+            <dpiAwareness>system, unaware</dpiAwareness>
```
[the remark this answers is not in the document]
Comment (ozone10) [reply PowerEditor/src/dpiAware.manifest:8]: `unaware` parameter is never used, on older Windows `dpiAwareness` is ignored and `dpiAware` is used, on newer `dpiAwareness` with "Per Monitor" parameters (if supported by OS) has higher priority.
 I included it mainly for hint how to add `PerMonitorV2` parameter in future 
```
<dpiAwareness>PerMonitorV2, system, unaware</dpiAwareness>
```
[ref](https://learn.microsoft.com/en-us/windows/win32/hidpi/setting-the-default-dpi-awareness-for-a-process)

**Comment by ozone10** [comment]
> If I do understand well, DPIManagerV2, which will manage better in Win11 and next version of Windows, can replace current DPI Manager in the future?

Yes, currently the old DPI Manager only support system aware context – it has same global dpi scale value for all windows/dialogs/controls/elements. But windows can have different dpi scale (e.g. find&replace dlg on second monitor) .

**Review by donho — APPROVED** [review]

**Comment by astewart77** [comment]
So is this the end of using n++ on Windows 7?

`The procedure entry point GetDpiForWindow could not be located in the dynamic link library USER32.dll. `

**Pull request closed without merging**